python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,469 @@
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import print_function, division
+
+import os.path
+import sys
+
+import numpy as np
+
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+from lsst.afw.table import SourceCatalog, SchemaMapper, Field
+from lsst.afw.table import MultiMatch, SimpleRecord, GroupView
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+# import lsst.afw.coord as afwCoord
+
+from .plotAstrometryPhotometry import plotAstrometry, plotPhotometry, plotPA1
+from .calcSrd import computeWidths, getRandomDiff, calcPA1, calcPA2
+
+def getCcdKeyName(dataid):
+    """Return the key in a dataId that's referring to the CCD or moral equivalent.
+
+    @param dataid  A dictionary that will be searched for a key that matches
+        an entry in the hardcoded list of possible names for the CCD field.
+
+    Different camera mappings use different keys
+      to indicate the different amps/ccds in the same exposure.
+    We here look through the reference dataId to determine which one this.
+    """
+    possibleCcdFieldNames = ['ccd', 'ccdnum', 'camcol']
+
+    for name in possibleCcdFieldNames:
+        if name in dataid:
+            return name
+    else:
+        return None
+
+
+def isExtended(source, extendedKey, extendedThreshold=1.0):
+    """Is the source extended attribute above the threshold.
+
+    Higher values of extendedness indicate a resolved object
+    that is larger than a point source.
+    """
+    return source.get(extendedKey) >= extendedThreshold
+
+def averageRaDec(cat):
+    """Calculate the RMS for RA, Dec for a set of observations an object."""
+    ra = np.mean(cat.get('coord_ra'))
+    dec = np.mean(cat.get('coord_dec'))
+    return ra, dec
+
+# Some thoughts from Paul Price on how to do the coordinate differences correctly:
+#    mean = sum([afwGeom.Extent3D(coord.toVector()) 
+#                for coord in coordList, afwGeom.Point3D(0, 0, 0)])
+#    mean /= len(coordList)
+#    mean = afwCoord.IcrsCoord(mean)
+
+def magNormDiff(cat):
+    """Calculate the normalized mag/mag_err difference from the mean for a 
+    set of observations of an objection.
+
+    @param[in]  cat -- Collection with a .get method 
+         for flux, flux+"-" 
+
+    @param[out]  pos_median -- median diff of positions in milliarcsec.  Float.
+    """
+    mag = cat.get('base_PsfFlux_mag')
+    magerr = cat.get('base_PsfFlux_magerr')
+    mag_avg = np.mean(mag)
+    N = len(mag)
+    normDiff = (mag - mag_avg) / magerr
+    
+
+
+def positionDiff(cat):
+    """Calculate the diff RA, Dec from the mean for a set of observations an object for each observation.
+
+    @param[in]  cat -- Collection with a .get method 
+         for 'coord_ra', 'coord_dec' that returns radians.
+
+    @param[out]  pos_median -- median diff of positions in milliarcsec.  Float.
+
+    This is WRONG!
```

Comment:
If it's wrong... why are we doing it? If it's just because it's good enough for now, can you file a ticket to indicate that it should be fixed in the future, rather than just leaving a warning in the code?